lib/appmap/value_inspector.rb

```diff
@@ -11,10 +12,17 @@ def detect_size(value)
       end
     end
 
-    def detect_schema(value, max_depth: MAX_DEPTH, type_info: {}, observed_values: Set.new(), depth: 0)
-      return type_info if depth == max_depth
+    def detect_schema(
+      value,
+      max_depth: MAX_DEPTH,
+      max_array_elements: MAX_ARRAY_ELEMENTS,
+      type_info: { class: best_class_name(value) },
+      observed_values: Set.new,
+      depth: 0
+    )
+      return type_info if depth >= max_depth && !array_like?(value)
```

Comment:
Not that it's a big issue, but it will dive deeply if there are Arrays of Arrays of Arrays of Arrays...

Reply:
I see there's a test for this later 👍

Reply:
It's possible, but it'll stop at the first `object` beyond or at `max_depth`.

So with a `max_depth` of `1`, this'll stop at the first object:
```
[ { children: [ { children: [ { children: [ { children: [ { children: [ { children: [ { children: []]]]]]]
```

The only real way to abuse this would be if you for some reason had deeply nested primitives, e.g.:
```
[[[[[[[[[[1]]]]]]]]]]
```